File: frontend/src/api/jobs.ts
import api from './client'
import type { Job } from '../types/job'

export async function listJobs(projectId?: string): Promise<Job[]> {
  const params = projectId ? { project_id: projectId } : {}
  const { data } = await api.get('/jobs', { params })
  return data
}

export async function getJob(jobId: string): Promise<Job> {
  const { data } = await api.get(`/jobs/${jobId}`)
  return data
}

File: backend/app/api/jobs.py
from fastapi import APIRouter, HTTPException

from app.api.deps import DbSession
from app.models.job import Job
from app.schemas.job import JobResponse

router = APIRouter(prefix="/jobs", tags=["jobs"])


@router.get("", response_model=list[JobResponse])
def list_jobs(db: DbSession, status: str | None = None, project_id: str | None = None):
    query = db.query(Job)
    if status:
        query = query.filter(Job.status == status)
    if project_id:
        query = query.filter(Job.project_id == project_id)
    return query.order_by(Job.created_at.desc()).all()


@router.get("/{job_id}", response_model=JobResponse)
def get_job(job_id: str, db: DbSession):
    job = db.query(Job).filter(Job.id == job_id).first()
    if not job:
        raise HTTPException(status_code=404, detail="Job not found")
    return job
